doc/source/rllib/core-concepts.rst

```diff
@@ -163,176 +163,162 @@ These batches are wrapped up together in a ``MultiAgentBatch``,
-      This ensures a fixed ratio of computations between, e.g., experience generation and experience replay.
-      The ratio can be adjusted by setting ``round_robin_weights``.
-    - ``async``: Execute each input dataflow as fast as possible without blocking.
-      You might want to use this when, e.g., you want replay to proceed as fast as possible irregardless of how fast experiences are being generated.
+**Concurrency ops**:
+RLlib provides a collection of concurrency ops that can be asynchronous and synchronous operations in the training loop.
+``AsyncRequestsManager`` is used for launching and managing asynchronous requests on actors. Currently in RLlib it is
+used for asynchronous sampling on rollout workers and asynchronously adding to and sampling from replay buffer actors.
+``synchronous_parallel_sample`` has a more narrow but common ussage of synchronously sampling from a set of rollout workers.
 
-**Metric ops** (`metric_ops.py <https://github.com/ray-project/ray/blob/master/rllib/execution/metric_ops.py>`__):
-Finally, we provide a ``StandardMetricsReporting`` operator that collects training metrics from the rollout workers in a unified fashion, and returns a stream of training result dicts.
-Execution plans should always end with this operator.
-This metrics op also reports various internal performance metrics stored by other operators in the shared metrics context accessible via ``_get_shared_metrics()``.
 
 Examples
 ~~~~~~~~
 
-.. dropdown::  **Example: Asynchrony**
+.. dropdown::  **Example: Synchronous Policy-Gradients**
 
-    Suppose we wanted to make the above A2C example asynchronous (i.e., A3C). We would switch the synchronous ``ParallelRollouts`` operator with ``AsyncGradients``, and use ``ApplyGradients`` to apply gradient updates as fast as they are collected. The ``AsyncGradients`` operator is going to execute rollouts in parallel, compute the policy gradient over the new batches (of size ``rollout_fragment_length``) on the remote workers, and then return a stream of the computed gradients:
+    This is what the Vanilla Policy Gradients algorithm looks like:
 
     .. code-block:: python
 
-        def execution_plan(workers: WorkerSet, config: TrainerConfigDict):
-            # type: LocalIterator[(ModelGradients, int)]
-            grads = AsyncGradients(workers)
-
-            # type: LocalIterator[_]
-            train_op = grads.for_each(ApplyGradients(workers, update_all=False))
-
-            # type: LocalIterator[ResultDict]
-            return StandardMetricsReporting(train_op, workers, config)
-
-    See also the `actual A3C implementation <https://github.com/ray-project/ray/blob/master/rllib/algorithms/a3c/a3c.py>`__.
-
-.. dropdown:: **Example: Replay**
-
-    Let's try adding a replay buffer to A2C. This can be done as follows by inserting store / replay ops and using ``Concurrently`` to compose them together:
+            def training_iteration(self) -> ResultDict:
```

Comment:
I think it would be best to reference the code itself here.
This way we can passively keep up with our own code changes.

.. literalinclude:: ../../../rllib/algorithms/a2c/a2c.py
    :language: python
    :start-after: __sphinx_doc_a2c_training_step__begin__
    :end-before: __sphinx_doc_a2c_training_step__end__
    
    Something like that. This will also need the __sphinx_doc_......__ markers inplace.

Reply:
removed the code example for now